### PR TITLE
DSOO-40-add-tests-to-the-project

### PR DESCRIPTION
**Pull request by SarahZimmermannSchmutzler**
**Description**

We want to add Tests to the Truck signs api project in a way, that:
- all models have at least 1 test
- all tests need to test for all exceptions/errors that could be raised
- all tests do need to have at least 1 test-Scenario with success.

**About the Tests**

The following bullet-points should help you getting started with testing and how to author tests.
- can I successfully create an object/model with all required properties?
      - can I successfully create an object with all required and optional properties
      - can I successfully create an object with a subset of required and optional properties
- Can I create a model with a missing required property? → If yes, this indicates a bug
- Does the model raise the correct Error when some information is missing in the creation?

**Acceptance Criteria**

- Each Model has one test for success
- Each Model has at least one test for failure
=> each model has 2+ tests

See the [docs](https://docs.djangoproject.com/en/5.2/topics/testing/overview/).

**Changes**
- src/tsa_products/tests.py
    - create sucess-test for creating a category
    - create schema for failure-test 
    - extend testing about setUp- and setUpData Functions
    - create tests (creating instances) for models: LetteringItemCategory, ProductColor, Product, Comment, Order, ProductVariation, LetterItemVariation
    - create test about methods for models: ProductVariation
    - create test folder and a file for every test class

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@ db.sqlite3-journal
 media
 mediafiles
 staticfiles
+src/static
```
Comment (spmse) [inline .gitignore]: remove the folder and re-run `collectstatic` command to recollect everything under staticfiles (which already is gitignored)

File: src/tsa_products/tests.py [hunk 2]
```diff
@@ -1,3 +1,335 @@
...
+    def setUpTestData(self):
+        # Creates testing data for the test cases.
+        self.test_category = Category.objects.create(title="test", image="test-path")
+        self.test_title = "test-title"
+        self.test_image = "test-path"
+        self.test_detail_image = "test-detail-path"
+        self.test_is_uploaded = False
+
+    def setUp(self):
+        # Ensures there are categories in the db before a test case is running.
```
Comment (spmse) [inline src/tsa_products/tests.py]: it rather ensure there are NO categories in the db, right?

File: src/tsa_products/tests.py [hunk 3]
```diff
@@ -1,3 +1,335 @@
...
+class LetteringItemCategoryTestCase(TestCase):
+
+    @classmethod
+    def setUpTestData(self):
+        # Creates testing data for the test cases.
+        self.test_title = "test"
+        self.test_price = 0.0
+
+    def setUp(self):
+        # Ensures there are categories in the db before a test case is running.
```
Comment (spmse) [inline src/tsa_products/tests.py]: it ensures all categories are deleted before running the test

File: src/tsa_products/tests.py [hunk 4]
```diff
@@ -1,3 +1,335 @@
...
+            product_variation.full_clean()
+            product_variation.save()
+        self.assertEqual(ProductVariation.objects.count(), 0)
+
+    def test_success_get_all_lettering_items_method(self):
+        test_product_variation = ProductVariation.objects.create(product=self.test_product)
+        item1 = LetteringItemVariation.objects.create(product_variation=test_product_variation)
+        item2 = LetteringItemVariation.objects.create(product_variation=test_product_variation)
+
+        items = test_product_variation.get_all_lettering_items()
```
Comment (spmse) [inline src/tsa_products/tests.py]: 1. would rename to `all_items`
2. (optional, if default sorting should be `by_id`) add a variable `sorted_items` that should be sorted by the IDs of the queryset values (like in the assert below)

File: src/tsa_products/tests.py [hunk 5]
```diff
@@ -1,3 +1,335 @@
...
+class ProductColorTestCase(TestCase):
+
+    @classmethod
+    def setUpTestData(self):
+        # Creates testing data for the test cases.
+        self.test_color_in_hex = "#000000"
+        self.test_color_nickname = "add nickname"
+
+    def setUp(self):
+        # Ensures there are categories in the db before a test case is running.
```
Comment (spmse) [inline src/tsa_products/tests.py]: same as for others

**Review by spmse — CHANGES_REQUESTED** [review]
Hi @SarahZimmermannSchmutzler ,
The PR already looks good - nice work of yours! I've left some last comments and change-requests before we can merge this
 
1. Refactor the code comments in the test functions (`def test_*`) in a way that they are docstrings
2. ensure to check all `setUp` functions to refactor the comments there to no longer be contradicting with your actual code
3. refactor your existing tests to add more assertions than just relying on db count info
    - check if the property values after adding the object to the database are correct (see the example comment in the review threads)
4. create a tests module (add a folder tests, that contains an empty `__init__.py` and multiple python files for the tests)    - the files should follow the naming pattern: `test_<ENTITY>.py` where entity should be replaced with the respective present model names, e.g. Category, Comment, etc.    - move all the test classes (test suites) from `tests.py` to the respective file in the new module
5. remove the `tests.py` file in the end

File: src/tsa_products/tests.py [hunk 6]
```diff
@@ -1,3 +1,59 @@
...
+
+    @classmethod
+    def setUpTestData(self):
+        # Creates testing data for the test cases.
+        self.test_title = "test"
+        self.test_image = "test-path"
+        self.test_base_price = 0.0
+        self.test_max_amount_of_lettering_items = -1
+        self.test_height = 0.0
+        self.test_width = 0.0
```
[the remark this answers is not in the document]
Comment (SarahZimmermannSchmutzler) [reply src/tsa_products/tests.py]: @spmse let's take that question into another ticket
